builtin/common/misc_helpers.lua

```diff
@@ -400,6 +400,80 @@ function core.explode_scrollbar_event(evt)
 	return retval
 end
 
+--------------------------------------------------------------------------------
+function core.explode_key_event(evt)
+	if evt == nil then
+		return nil
+	end
+
+	local parts = evt:split(";", true)
+	if #parts < 3 then
+		return nil
+	end
+
+	local ret = {
+		key = parts[1],
+		pressed = parts[3] == "true"
+	}
+
+	if ret.key == "Semicolon" then
+		ret.key = ";"
+	end
+
+	if parts[2] ~= "" then
+		ret.mapped = parts[2]
+	end
+
+	return ret
+end
+
+--------------------------------------------------------------------------------
+function core.explode_mouse_event(evt)
+	if evt == nil then
+		return nil
+	end
+
+	local parts = evt:split(";", true)
```

Comment:
At this point I'm wondering why you don't just send events as JSON.

Reply:
I tried that before.  `minetest.parse_json()` prints bunches of errors on invalid JSON or something like that, I forget the exact reason why I eventually decided against it.

Reply:
`minetest.parse_json` is indeed crap [as documented here](https://github.com/minetest/minetest_docs/pull/13) but should suffice for this narrow use case